frontend/src/app/pages/DashBoardPage/utils/__tests__/board.test.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Datart
+ *
+ * Copyright 2021
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Widget } from '../../pages/Board/slice/types';
+import { adjustWidgetsBeforeSave } from '../board';
+
+describe('board editor events', () => {
+  test('adjustWidgetsBeforeSave', () => {
```

Comment:
It is better to give more business info for test description. such as `should get new target version after adjust widgets before save`

Reply:
nice